src/fides/api/schemas/connection_configuration/connection_secrets_s3.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List, Optional
+
+from pydantic import Field, root_validator
+
+from fides.api.schemas.base_class import NoValidationSchema
+from fides.api.schemas.connection_configuration.connection_secrets import (
+    ConnectionConfigSecretsSchema,
+)
+from fides.api.schemas.storage.storage import AWSAuthMethod
+
+
+class S3Schema(ConnectionConfigSecretsSchema):
+    """Schema to validate the secrets needed to connect to Amazon S3"""
+
+    auth_method: AWSAuthMethod = Field(
+        title="Authentication Method",
+        description="Determines which type of authentication method to use for connecting to Amazon S3",
+    )
+
+    aws_access_key_id: Optional[str] = Field(
+        title="Access Key ID",
+        description="Part of the credentials that provide access to your AWS account. This is required if using secret key authentication.",
+    )
+    aws_secret_access_key: Optional[str] = Field(
+        title="Secret Access Key",
+        description="Part of the credentials that provide access to your AWS account. This is required if using secret key authentication.",
+        sensitive=True,
+    )
+
+    aws_assume_role_arn: Optional[str] = Field(
+        title="Assume Role ARN",
+        description="If provided, the ARN of the role that should be assumed to connect to s3.",
+    )
```

Comment:
Schema looks good! From my perspective:

- Access Key ID: not particularly sensitive, a bit like a username
- Secret Access Key: sensitive, treat it like a pssword
- Assume Role ARN: not particularly sensitive, again a bit like a username